src/components/DateRangePicker.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+import React from "react";
+import { Calendar } from "react-native-calendars";
+import formatDate from "date-fns/format";
+import addDays from "date-fns/add_days";
+import isBefore from "date-fns/is_before";
+import { cardBgColor, eventListHeaderColor } from "../constants/colors";
+import type { DateRange, DateOrDateRange } from "../data/date-time";
+
+type CalendarDay = {
+  year: number,
+  month: number,
+  day: number,
+  timestamp: number,
+  dateString: string
+};
+
+const getSortedDateRange = (dates: DateRange) => {
+  if (dates.startDate === dates.endDate) {
+    return dates.startDate;
+  }
+
+  return isBefore(dates.endDate, dates.startDate)
+    ? { startDate: dates.endDate, endDate: dates.startDate }
+    : dates;
+};
+
+const getMarkedDate = (date: string) => ({
+  [date]: {
+    selected: true,
+    selectedColor: eventListHeaderColor
+  }
+});
+
+const getMarkedDateRange = (dateRange: DateRange) => {
+  const { startDate, endDate } = dateRange;
+  const template = {
+    color: eventListHeaderColor,
+    textColor: cardBgColor
+  };
+
+  const markedDates = {};
+
+  // Start
+  markedDates[startDate] = {
+    ...template,
+    startingDay: true
+  };
+
+  // In between
+  let inBetweenDate = startDate;
+  do {
+    inBetweenDate = formatDate(addDays(inBetweenDate, 1), "YYYY-MM-DD");
+    markedDates[inBetweenDate] = { ...template };
```

Comment:
If I'm reading it right, `markedDates` are immutable, so you could just reuse the `template` object here, without the spread. Not that it matters :)

Reply:
We can absolutely reuse template here 🙂